### PR TITLE
Rework flash subsystem

### PR DESCRIPTION
**Pull request by Deryugin**
* Simplify interface
* Rework flash emulator
* Fix DumbFS
* Fix block devices idesc ops for new VFS

### CODE REVIEW

File: src/cmds/fs/mkflashemu.c [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+static void print_usage(void) {
+	printf("Usage: mkflashemu [-h] [[-n BLOCKS] [-b BLOCK_SIZE] dev_name]\n");
+}
+
+int main(int argc, char **argv) {
+	int opt;
+	size_t block_size = 512;
+	int blocks = 1024;
+
+	getopt_init();
```
Comment (anton-bondarev) [inline src/cmds/fs/mkflashemu.c]: it's not necessary

File: src/drivers/flash/flash.c [hunk 2]
```diff
@@ -1,328 +1,132 @@
...
-
-static int flashbdev_write(struct block_dev *bdev,
-		char *buffer, size_t count, blkno_t blkno) {
-	struct flash_dev *flash;
-	uint32_t startpos, endpos;
-
-	flash = (struct flash_dev *)bdev->privdata;
-	startpos = flash->start + (blkno * flash->block_info.block_size);
-	endpos = startpos + count - 1;
+static int flash_initialized = 0;
```
Comment (anton-bondarev) [inline src/drivers/flash/flash.c]: May be use UNIT_INIT?

File: src/cmds/fs/mkflashemu.c (hunk 1 above)
Comment (Deryugin) [reply src/cmds/fs/mkflashemu.c]: fixed

File: src/drivers/flash/flash.c (hunk 2 above)
Comment (Deryugin) [reply src/drivers/flash/flash.c]: This is necessary due to broken RunLevel sequence: sometimes we have to init devices before init of this module.

**Pull request merged**